tests/Unit/Domain/Test_AreaElement.cpp

```diff
@@ -0,0 +1,230 @@
+    const auto& initial_ref_levels = shell.initial_refinement_levels();
+    const auto element_ids = initial_element_ids(initial_ref_levels);
+    const auto excision_sphere =
+        shell_domain.excision_spheres().at("ExcisionSphere");
+    const Mesh<2> face_mesh{10, Spectral::Basis::Legendre,
+                            Spectral::Quadrature::GaussLobatto};
+    domain::CoordinateMaps::KerrHorizonConforming kerr_schild_map(mass, spin);
+
+    // Shell does not distort to conform to the horizon in Kerr-Schild
+    // coordinates for spinning black holes, so we manually append the horizon
+    // conforming map
+    const auto grid_to_kerr_conforming_map =
+        CoordinateMap<Frame::Inertial, Frame::Inertial,
+                      domain::CoordinateMaps::KerrHorizonConforming>(
+            kerr_schild_map);
+    gr::Solutions::KerrSchild kerr_schild{mass, spin, {0., 0., 0.}};
+    double horizon_area = 0.;
+    for (const auto& element_id : element_ids) {
+      const auto& current_block = blocks.at(element_id.block_id());
+      const auto element_abutting_direction =
+          excision_sphere.abutting_direction(element_id);
+      if (element_abutting_direction.has_value()) {
+        const auto face_logical_coords = interface_logical_coordinates(
+            face_mesh, element_abutting_direction.value());
+        const ElementMap logical_to_grid_map(
+            element_id, current_block.stationary_map().get_clone());
+        const auto inertial_coords = logical_to_grid_map(face_logical_coords);
+        const auto inv_jacobian =
+            logical_to_grid_map.inv_jacobian(face_logical_coords);
+        const auto inv_jacobian_kerr_horizon =
+            grid_to_kerr_conforming_map.inv_jacobian(inertial_coords);
+        const auto total_inv_jacobian = tenex::evaluate<ti::I, ti::j>(
+            inv_jacobian(ti::I, ti::k) *
+            inv_jacobian_kerr_horizon(ti::K, ti::j));
+        const auto spacetime_vars = kerr_schild.variables(
+            grid_to_kerr_conforming_map(inertial_coords), 0.,
+            tmpl::list<
+                gr::Tags::InverseSpatialMetric<3, Frame::Inertial, DataVector>,
+                gr::Tags::SqrtDetSpatialMetric<DataVector>>{});
+        const auto& inverse_spatial_metric =
+            get<gr::Tags::InverseSpatialMetric<3, Frame::Inertial, DataVector>>(
+                spacetime_vars);
+        const auto curved_area_element = area_element(
+            total_inv_jacobian, element_abutting_direction.value(),
+            inverse_spatial_metric,
+            get<gr::Tags::SqrtDetSpatialMetric<DataVector>>(spacetime_vars));
+        horizon_area += definite_integral(get(curved_area_element), face_mesh);
+      }
+    }
+    auto custom_approx = Approx::custom().epsilon(1e-8);
+    CHECK(horizon_area == custom_approx(8. * M_PI * mass * horizon_radius));
+  }
+}
+
+SPECTRE_TEST_CASE("Unit.Domain.AreaElement", "[Domain][Unit]") {
+  test_area_element_against_normal_vector();
```

Comment:
Why did you remove the grid frame here?

Reply:
the `Minkowski` variables method is hard-coded on the inertial frame. I can template it if you think it's necessary.

Reply:
no this is fine